sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java

```diff
@@ -217,15 +222,35 @@ private void captureRequestHeaders(HttpHeaders requestHeaders, Map<String, Strin
         }
     }
 
-    private String redactUserDelegationKey(String content) {
-        if (!content.contains("UserDelegationKey")) {
-            return content;
+    private String redactSensitiveContent(String content) {
+        // replace all sensitive string key in the black list
+        for (String blackString : BLACK_LIST) {
```

Comment:
It'll be more overhead but I think turning the stopword `String` into a new class, such as `RecordingRedactor`, where it contains the stopword and a function to redact would clean the code up a lot and make it more easily extended as new requests come in.

```java
class RecordingRedactor {
  private final String stopword;

  public boolean shouldRedact(String redactableString);
  public String redact(String redactableString);
}
```

Reply:
updated